### PR TITLE
added dont_use_soft_clipped_bases option

### PR DESCRIPTION
**Pull request by pcantalupo**
# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by MaxUlysse** [comment]
Hi @pcantalupo, Thanks a lot for your PR.
I have a couple issues.
It seems that you made it from the `master` branch instead of the `dev` branch.
I changed the target branch to `dev` and we now have way more modifications that we should have.
I'll suggest you to follow our [contributing guidelines](https://github.com/nf-core/sarek/blob/master/.github/CONTRIBUTING.md).

**Comment by drpatelh** [comment]
`--ignore_soft_clipped_bases` is better I think. Could use `--ignore_soft_clipped` too but that may lead to confusion with reads that are soft-clipped and not the bases themselves.

File: main.nf [hunk 1]
```diff
@@ -2465,6 +2466,7 @@ process Mutect2 {
     // https://gatkforums.broadinstitute.org/gatk/discussion/11136/how-to-call-somatic-mutations-using-gatk4-mutect2
     PON = params.pon ? "--panel-of-normals ${pon}" : ""
     intervalsOptions = params.no_intervals ? "" : "-L ${intervalBed}"
+    softclippedOption = params.dont_use_soft_clipped_bases ? "--dont-use-soft-clipped-bases true" : ""
```
Comment (FriederikeHanssen) [inline main.nf:2469]: I would prefer camel case here for readability  `softClippedOption`

File: main.nf [hunk 2]
```diff
@@ -2473,6 +2475,7 @@ process Mutect2 {
       -I ${bamTumor}  -tumor ${idSampleTumor} \
       -I ${bamNormal} -normal ${idSampleNormal} \
       ${intervalsOptions} \
+      ${softclippedOption} \
```
Comment (FriederikeHanssen) [inline main.nf:2478]: see above

File: main.nf [hunk 3]
```diff
@@ -94,6 +94,7 @@ def helpMessage() {
                                       Default: ${params.cf_ploidy}
       --cf_window               [int] Control-FREEC window size
                                       Default: Disabled
+      --dont_use_soft_clipped_bases [bool] Do not analyze soft clipped bases in the reads for GATK Mutect2
```
Comment (MaxUlysse) [inline main.nf:97]:
```suggestion
      --ignore_soft_clipped_bases [bool] Do not analyze soft clipped bases in the reads for GATK Mutect2
```

File: main.nf (hunk 1 above)
Comment (MaxUlysse) [inline main.nf:2469]:
```suggestion
    softClippedOption = params.ignore_soft_clipped_bases ? "--dont-use-soft-clipped-bases true" : ""
```

File: main.nf (hunk 2 above)
Comment (MaxUlysse) [inline main.nf:2478]:
```suggestion
      ${softClippedOption} \
```

File: nextflow.config [hunk 4]
```diff
@@ -66,6 +66,7 @@ params {
   no_strelka_bp = null // Strelka will use Manta candidateSmallIndels if available
   pon = false // No default PON (Panel of Normals) file for GATK Mutect2 / Sentieon TNscope
   pon_index = false // No default PON index for GATK Mutect2 / Sentieon TNscope
+  dont_use_soft_clipped_bases = null // GATK Mutect2
```
Comment (MaxUlysse) [inline nextflow.config:69]:
```suggestion
  ignore_soft_clipped_bases = null // GATK Mutect2
```

**Comment by MaxUlysse** [comment]
@pcantalupo if you agree with the suggestions we propose, you can batch accept them if you're in the files changes tab.
That way you can do only one commit directly here on github

**Comment by MaxUlysse** [comment]
Thanks a lot @pcantalupo for this first PR

**Pull request merged**